src/Symfony/Bundle/SecurityBundle/Resources/config/security.php

```diff
@@ -0,0 +1,276 @@
+
+        ->set('security.role_hierarchy', RoleHierarchy::class)
+            ->args([param('security.role_hierarchy.roles')])
+        ->alias(RoleHierarchyInterface::class, 'security.role_hierarchy')
+
+        // Security Voters
+        ->set('security.access.simple_role_voter', RoleVoter::class)
+            ->tag('security.voter', ['priority' => 245])
+
+        ->set('security.access.authenticated_voter', AuthenticatedVoter::class)
+            ->args([service('security.authentication.trust_resolver')])
+            ->tag('security.voter', ['priority' => 250])
+
+        ->set('security.access.role_hierarchy_voter', RoleHierarchyVoter::class)
+            ->args([service('security.role_hierarchy')])
+            ->tag('security.voter', ['priority' => 245])
+
+        ->set('security.access.expression_voter', ExpressionVoter::class)
+            ->args([
+                service('security.expression_language'),
+                service('security.authentication.trust_resolver'),
+                service('security.authorization_checker'),
+                service('security.role_hierarchy')->nullOnInvalid()
+            ])
+            ->tag('security.voter', ['priority' => 245])
+
+        // Firewall related services
+        ->set('security.firewall', FirewallListener::class)
+            ->tag('kernel.event_subscriber')
+            ->args([
+                service('security.firewall.map'),
+                service('event_dispatcher'),
+                service('security.logout_url_generator'),
+            ])
+        ->alias(Firewall::class, 'security.firewall')
+
+        ->set('security.firewall.map', FirewallMap::class)
+            ->args([
+                abstract_arg('Firewall context locator'),
+                abstract_arg('Request matchers'),
+            ])
+
+        ->set('security.firewall.context', FirewallContext::class)
+            ->abstract()
+            ->args([
+                [],
+                service('security.exception_listener'),
+                abstract_arg('LogoutListener'),
+                abstract_arg('FirewallConfig'),
+            ])
+
+        ->set('security.firewall.lazy_context', LazyFirewallContext::class)
+            ->abstract()
+            ->args([
+                [],
+                service('security.exception_listener'),
+                abstract_arg('LogoutListener'),
+                abstract_arg('FirewallConfig'),
+                service('security.untracked_token_storage'),
+            ])
+
+        ->set('security.firewall.config', FirewallConfig::class)
+            ->abstract()
+            ->args([
+                '',
+                abstract_arg('user_checker'),
+                abstract_arg('request_matcher'),
+                false, // security enabled
+                false, // stateless
+                null,
+                null,
+                null,
+                null,
+                null,
```

Comment:
Shouldn't we use `abstract_arg` instead ?

Reply:
argument 6 of "Symfony\Bundle\SecurityBundle\Security\FirewallConfig::__construct" accepts "string"_

i don't know why but on lines +204 to +208 should be null and abstract_arg failed.

Reply:
maybe default values ?